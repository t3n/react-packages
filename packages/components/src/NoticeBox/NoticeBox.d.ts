import React from 'react';
import { SpaceProps } from 'styled-system';
export declare const NoticeBoxText: React.FC;
declare const NoticeBox: React.FC<SpaceProps>;
export default NoticeBox;
