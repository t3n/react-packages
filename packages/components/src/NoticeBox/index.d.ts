import NoticeBox from './NoticeBox';
export * from './NoticeBox';
export default NoticeBox;
