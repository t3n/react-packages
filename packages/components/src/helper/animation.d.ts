import { ThemeProps } from '@t3n/theme';
export declare const backgroundAnimation: import("styled-components").FlattenInterpolation<import("styled-components").ThemedStyledProps<ThemeProps, any>>;
