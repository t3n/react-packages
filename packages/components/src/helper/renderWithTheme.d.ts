import 'jest-styled-components';
import React from 'react';
import { RenderOptions } from '@testing-library/react';
import { Theme } from '@t3n/theme';
interface OptionalTheme {
    theme?: Theme;
}
export declare const renderWithTheme: (ui: React.ReactElement, { theme, ...options }: RenderOptions & OptionalTheme) => import("@testing-library/react").RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement>;
export {};
