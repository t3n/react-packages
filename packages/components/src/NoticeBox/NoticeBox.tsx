import React from 'react';
import styled from 'styled-components';
import { border, color, space, SpaceProps } from 'styled-system';

import Box from '../Box';
import Text from '../Text';

const StyledNoticeBox = styled(Box)`
  ${({ theme }) =>
    border({
      theme,
      borderColor: 'shades.grey42',
      borderWidth: '1px 2px 4px 1px',
      borderStyle: 'solid',
      borderRadius: theme.border.radii[1],
    })}
  ${({ theme }) => color({ theme, bg: 'feedback.notice' })}

  ${space};
`;

export const NoticeBoxText: React.FC = ({ children }) => {
  return (
    <Text m={0} width="100%">
      {children}
    </Text>
  );
};

const NoticeBox: React.FC<SpaceProps> = ({ children, ...rest }) => {
  return (
    <StyledNoticeBox display="flex" alignItems="center" p={3} {...rest}>
      {children}
    </StyledNoticeBox>
  );
};

export default NoticeBox;
