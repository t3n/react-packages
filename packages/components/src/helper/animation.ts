import { css, keyframes } from 'styled-components';

import { ThemeProps } from '@t3n/theme';

const backgroundAnimationKeyframes = ({ theme }: ThemeProps) => keyframes`
  0% {
    background-color: ${theme.colors.shades.grey204}
  }

  50% {
    background-color: ${theme.colors.shades.grey244}
  }

  100% {
    background-color: ${theme.colors.shades.grey204}
  }
`;

// eslint-disable-next-line import/prefer-default-export
export const backgroundAnimation = css`
  animation: ${backgroundAnimationKeyframes} 2s linear infinite;
`;
