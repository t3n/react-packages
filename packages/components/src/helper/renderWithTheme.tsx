/* eslint-disable import/no-extraneous-dependencies */
import 'jest-styled-components';

import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { Theme, theme as originalTheme } from '@t3n/theme';

interface OptionalTheme {
  theme?: Theme;
}

// eslint-disable-next-line import/prefer-default-export
export const renderWithTheme = (
  ui: React.ReactElement,
  { theme, ...options }: RenderOptions & OptionalTheme
) => {
  // merge over default theme

  const mergedTheme = { ...originalTheme, ...theme };

  const Wrapper: React.FC = ({ children }) => (
    <ThemeProvider theme={mergedTheme}>{children}</ThemeProvider>
  );

  return render(ui, { wrapper: Wrapper, ...options });
};
