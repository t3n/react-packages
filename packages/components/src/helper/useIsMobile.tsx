import { Theme } from '@t3n/theme';
import { useCallback, useContext, useEffect, useState } from 'react';
import { ThemeContext } from 'styled-components';

const useIsMobile = (): boolean => {
  const [isMobile, setIsMobile] = useState(false);

  const theme: Theme = useContext(ThemeContext);
  const fontSize = window
    .getComputedStyle(window.document.body)
    .getPropertyValue('font-size');
  const mobileBreakpoint =
    parseInt(theme.breakpoints[1], 10) * parseInt(fontSize, 10);

  const testIsMobile = useCallback(() => {
    if (window.matchMedia(`(max-width: ${mobileBreakpoint}px)`).matches) {
      setIsMobile(true);
    } else {
      setIsMobile(false);
    }
  }, [mobileBreakpoint]);

  useEffect(() => {
    testIsMobile();
    window.addEventListener('resize', testIsMobile);

    return () => {
      window.removeEventListener('resize', testIsMobile);
    };
  }, [testIsMobile]);

  return isMobile;
};

export default useIsMobile;
