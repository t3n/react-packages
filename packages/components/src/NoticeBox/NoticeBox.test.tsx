import React from 'react';

import { renderWithTheme } from '../helper/renderWithTheme';
import NoticeBox, { NoticeBoxText } from './NoticeBox';

test('NoticeBox Status matches Snapshot', () => {
  const { container } = renderWithTheme(
    <NoticeBox m={2}>
      <NoticeBoxText>Dies ist eine Erfolgsmeldung</NoticeBoxText>
    </NoticeBox>,
    {}
  );

  expect(container).toMatchSnapshot();
});
