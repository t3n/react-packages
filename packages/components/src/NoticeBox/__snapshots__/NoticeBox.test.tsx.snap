// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeBox Status matches Snapshot 1`] = `
.c0 {
  padding: 16px;
  margin: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  font-family: -apple-system,system-UI,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,sans-serif;
  font-size: 1rem;
  line-height: 1.5em;
  margin: 0;
  width: 100%;
}

.c1 {
  border-color: #2a2a2a;
  border-width: 1px 2px 4px 1px;
  border-style: solid;
  border-radius: 4px;
  background-color: #56FEFF;
  padding: 16px;
  margin: 8px;
}

.c1 a {
  background-image: url('data:image/svg+xml;base64,PHN2ZyBwcmVzZXJ2ZUFzcGVjdFJhdGlvPSJub25lIiB2aWV3Qm94PSIwIDAgMSAxIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIHN0cm9rZT0icmdiKDk1LDk1LDk1KSI+PHJlY3QgeD0iMCIgeT0iMCIgd2lkdGg9IjEiIGhlaWdodD0iMSIgLz48L2c+PC9zdmc+');
}

.c1 a:visited,
.c1 a:hover,
.c1 a:focus {
  color: #2a2a2a;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-repeat: repeat-x;
  background-size: 1px 1px;
  background-position: 0 100%;
  background-image: url('data:image/svg+xml;base64,PHN2ZyBwcmVzZXJ2ZUFzcGVjdFJhdGlvPSJub25lIiB2aWV3Qm94PSIwIDAgMSAxIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIHN0cm9rZT0icmdiKDI0OSw2Niw1OCkiPjxyZWN0IHg9IjAiIHk9IjAiIHdpZHRoPSIxIiBoZWlnaHQ9IjEiIC8+PC9nPjwvc3ZnPg==');
}

@media screen and (min-width:30rem) {
  .c2 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:48rem) {
  .c2 {
    line-height: 1.666em;
  }
}

<div>
  <div
    class="c0 c1"
    display="flex"
  >
    <p
      class="c2"
      width="100%"
    >
      Dies ist eine Erfolgsmeldung
    </p>
  </div>
</div>
`;
